src/classlibnative/bcltype/system.cpp

```diff
@@ -41,14 +41,35 @@ typedef BOOL (*pfnGetPhoneVersion)(LPOSVERSIONINFO lpVersionInformation);
 pfnGetPhoneVersion g_pfnGetPhoneVersion = NULL;
 #endif
 
+typedef void(WINAPI *pfnGetSystemTimeAsFileTime)(LPFILETIME lpSystemTimeAsFileTime);
+pfnGetSystemTimeAsFileTime g_pfnGetSystemTimeAsFileTime = NULL;
+Volatile<bool> g_fGetSystemTimeAsFileTimeInitialized = false;
 
 FCIMPL0(INT64, SystemNative::__GetSystemTimeAsFileTime)
 {
     FCALL_CONTRACT;
 
-    INT64 timestamp;
+	if (!g_fGetSystemTimeAsFileTimeInitialized)
+	{
+		// Try to use GetSystemTimePreciseAsFileTime if it's available (Win8+).
+		// Otherwise fall back to GetSystemTimeAsFileTime.
+		pfnGetSystemTimeAsFileTime func = NULL;
+		HMODULE hKernel32 = WszLoadLibrary(W("kernel32.dll"));
```

Comment:
This should be under `#ifndef FEATURE_PAL` so that we are not trying to load kernel32.dll on Unix.